cli/ops/jupyter.rs

```diff
@@ -30,6 +32,61 @@ deno_core::extension!(deno_jupyter,
   },
 );
 
+#[op2(async)]
+#[string]
+pub async fn op_jupyter_input(
+  state: Rc<RefCell<OpState>>,
+  #[string] prompt: String,
+  #[serde] is_password: serde_json::Value,
+) -> Result<Option<String>, AnyError> {
+  let (_iopub_socket, last_execution_request, stdin_socket) = {
+    let s = state.borrow();
+
+    (
+      s.borrow::<Arc<Mutex<Connection<zeromq::PubSocket>>>>()
+        .clone(),
+      s.borrow::<Rc<RefCell<Option<JupyterMessage>>>>().clone(),
+      s.borrow::<Arc<Mutex<Connection<zeromq::RouterSocket>>>>()
+        .clone(),
+    )
+  };
+
+  let mut stdin = stdin_socket.lock().await;
+
+  let maybe_last_request = last_execution_request.borrow().clone();
+  if let Some(last_request) = maybe_last_request {
+    if !last_request.allow_stdin() {
+      return Ok(None);
+    }
+
+    /*
+     * Using with_identities() because of jupyter client docs instruction
+     * Requires cloning identities per :
+     * https://jupyter-client.readthedocs.io/en/latest/messaging.html#messages-on-the-stdin-router-dealer-channel
+     * The stdin socket of the client is required to have the
+     *  same zmq IDENTITY as the client’s shell socket.
+     *  Because of this, the input_request must be sent with the same IDENTITY
+     *  routing prefix as the execute_reply in order for the frontend to receive the message.
+     * """
+     */
+    last_request
```

Comment:
AFAIK, I think this is correct.

Reply:
Thank you 👏

Reply:
Confirmed this by the way and incorporated it directly into the new library to make it easier to do.